### ISSUE TITLE
mask or segmentation

### ISSUE THREAD

**Issue by sara-kassani** [GalAvineri/ISIC-Archive-Downloader#9]
How can I download dataset mask or segmentation images?

**Comment by GalAvineri** [comment]
Currently this option isn't available, but this is a great idea for a new feature :)

**Comment by neherh** [comment]
I just wanted to follow up on this issue. Has this new feature been provided?

**Comment by GalAvineri** [comment]
Not yet but it is nearly finished.
I hope that a in a few days i'll be able to finish it :)

**Comment by neherh** [comment]
That is awesome. Do you by chance know if all of the images have associated masks?

**Comment by GalAvineri** [comment]
I've added the ability to download the segmentation of images as well.
You can also choose do download only the segmentation of the images, without the images themselves.

To download segmentations you can do the following:
`python download_archive.py -s`
To skip downloading the images as well (if perhaps you've already downloaded them to this point)
you can use this flag:
`python download_archive.py -s --no-images`

Regarding you question :)
I do not know whether all the images have segmentation, though it's an interesting question which i'd like to know the answer to as well :P

**Comment by sara-kassani** [comment]
Awesome news.
Thanks!

**Comment by GalAvineri** [comment]
I'll be closing this than :)
If any issues arise with this new features, let me know!

**Comment by GalAvineri** [comment]
@neherh you're question actually discovered a bug in my code
If you will try to download the segmentation of an image that does not have one,
the program would be stuck on that image.

I'll try to solve as soon as i can :)

**Comment by neherh** [comment]
Perfect :) I am glad my question helped. I am glad you made (and continuing) the segmentation because it is not intuitive to download the segmentation using rest api; I couldn't figure it out. 

I do have one other question, there are some images that have 2 segmentations where a novice and expert provided different segmentations, are they both downloaded as well?

**Comment by GalAvineri** [comment]
Oh! I wasn't aware of that
Can you give an example of an image so that i can look into it?

**Comment by GalAvineri** [comment]
That example provided me with all the information i needed to be able to code it.
I'll be sure to include this option as well, thanks!

**Comment by neherh** [comment]
@GalAvineri does your code now support downloading both segmentation masks? With that being said, can this issue now be closed?

Thanks.

**Comment by GalAvineri** [comment]
The latest version of the code does support download of segmentation masks.
You should now be able to get a single segmentation mask for every image that has segmentation masks available.

But - there's still an issue with this functionallity.
Some images has multiple segmentation masks, which may vary on skill level.
The system currently downloads one of the masks, without consideration of the skill level.

I have kept this issue open because of this improvement that needs to be made, but i think that instead it should be represented as an independent issue.

So in conclusion :)
1. The code supports segmentation masks downloading
2. It still needs more improvments
3. This issue can be closed

If you're having any issues with the current version of the segmentation masks downloading, don't hesitate to reopen this issue or open a new one!

**Comment by LittleWhale0531** [comment]
hi，I have some troubles when downloading Segmentations.  The problem is that when the offset is greater than a certain value,I cannot download the segmentations.

For example, I execute ‘python download_archive.py --num-images 50 -s --seg-skill expert --offset 13000’ .As a result, only descriptions and images can be downloaded, but segmentation is not downloaded.

I found that there is no corresponding segmentation id in the ISIC API.
For example, the id of image 'ISIC_000000' is 5436e3abbae478396759f0cf, then I can use the following url to get the id of the corresponding segmentation。
![image](https://user-images.githubusercontent.com/46144065/56470459-ead21380-6478-11e9-8a33-c195f0573619.png)

The id of ISIC_0028595 is 5aaf13f01165976913635462, and there is no way to get the id of the corresponding segmentation with the following url.
![image](https://user-images.githubusercontent.com/46144065/56470465-fcb3b680-6478-11e9-8313-ad34c11c9ff9.png)

I want to download the segmentation of all images. How can I solve this problem?
thank you very much!
